prov/util/src/util_cq.c

```diff
@@ -584,37 +584,50 @@ int ofi_check_bind_cq_flags(struct util_ep *ep, struct util_cq *cq,
 	return FI_SUCCESS;
 }
 
-void ofi_cq_progress(struct util_cq *cq)
+static inline void ofi_cq_progress(struct util_cq *cq)
 {
 	struct util_ep *ep;
 	struct fid_list_entry *fid_entry;
 	struct dlist_entry *item;
 
-	cq->cq_fastlock_acquire(&cq->ep_list_lock);
 	dlist_foreach(&cq->ep_list, item) {
 		fid_entry = container_of(item, struct fid_list_entry, entry);
 		ep = container_of(fid_entry->fid, struct util_ep, ep_fid.fid);
 		ep->progress(ep);
 
 	}
+}
+
+void ofi_cq_progress_ts(struct util_cq *cq)
+{
+	cq->cq_fastlock_acquire(&cq->ep_list_lock);
```

Comment:
Should we use `fastlock_acquire/release` instead?

Reply:
This was just moving the existing code, but I agree.  There's no need for the cq locking functions with this change.  I'll prepare a follow on patch that removes them.